changelog/upgrade-notes.md

```diff
@@ -2,6 +2,13 @@
 
 ### v3.5.0
 
+#### `ol.tilegrid` changes
+
+* If you previously created an XYZ tilegrid, you could construct an `ol.tilegrid.XYZ`. This class has been replaced by a static `ol.tilegrid.createXYZ` function which takes the same options as the previous `ol.tilegrid.XYZ` constructor, but returns an `ol.tilegrid.TileGrid` instance. The internal tile coordinate scheme for XYZ tile grids has been changed. Previously, the `y` of tile coordinates was transformed to the coordinates used by sources by calculating `-y-1`. Now, it is transformed by calculating `height-y-1`, where height is the number of rows of the tile grid at the zoom level of the tile coordinate.
+* The `widths` constructor option of `ol.tilegrid.TileGrid` and subclasses is no longer available, and it is no longer necessary to get proper wrapping at the 180° meridian. However, for `ol.tilegrid.WMTS`, there is a new option `sizes`, where each entry is an `ol.Size` with the `width` ('TileMatrixWidth' in WMTS capabilities) as first and the `height` ('TileMatrixHeight') as second entry of the array. For other tile grids, users can
+now specify an `extent` instead of `widths´. These settings are used to restrict the range of tiles that sources will request.
+* For `ol.source.TileWMS`, the default value of `warpX` used to be `undefined`, meaning that WMS requests with out-of-extent tile BBOXes would be sent. Now `wrapX` can only be `true` or `false`, and the new default is `true`. No application code changes should be required, but the resulting WMS requests for out-of-extent tiles will be different.
```

Comment:
I think this deserves a bit more explanations. For example, how will the WMS requests for out-of-extent tiles will be different?